test/integration/react-profiling-mode/test/index.test.js

```diff
@@ -30,29 +41,37 @@ describe('React Profiling Mode', () => {
 
   describe('with config enabled', () => {
     beforeAll(async () => {
-      await fs.writeFile(
-        nextConfig,
-        `
-        module.exports = {
-          reactProductionProfiling: true
-        }
-      `
-      )
       await nextBuild(appDir, ['--profile'])
       appPort = await findPort()
       app = await nextStart(appDir, appPort)
     })
     afterAll(async () => {
-      await fs.remove(nextConfig)
       await killApp(app)
     })
 
-    it('should have used the react-dom profiling bundle', async () => {
+    it('should have used the react-dom profiling bundle for pages', async () => {
       const browser = await webdriver(appPort, '/')
       const results = await browser.eval('window.profileResults')
 
       expect(results.length).toBe(1)
       expect(results[0] && results[0][0]).toBe('hello')
     })
+
+    it('should have used the react-dom profiling bundle for client component', async () => {
+      const browser = await webdriver(appPort, '/client')
+      const results = await browser.eval('window.profileResults')
```

Comment:
Is this available immediately or should we use `check()` here?

Reply:
in the above pages test looks like available immediately

>The Profiler requires an onRender function as a prop. React calls this function any time a component within the profiled tree “commits” an update. It receives parameters describing what was rendered and how long it took.